samples/gallery/src/main/java/com/bumptech/glide/samples/gallery/HorizontalGalleryFragment.kt

```diff
@@ -1,50 +1,559 @@
+          sizeReadyCallbacks.remove(cb)
         }
       }
+      Log.d("TEST", "Start load for $model into $target")
+      requestBuilder.into(target)
+      awaitClose {
+        Log.d("TEST", "Close $model")
+        requestManager.clear(target)
+      }
     }
   }
 
-  override fun onCreateView(
-    inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?,
-  ): View? {
-    val result = inflater.inflate(R.layout.recycler_view, container, false)
-    recyclerView = result.findViewById<View>(R.id.recycler_view) as RecyclerView
-    val layoutManager = GridLayoutManager(activity, 1)
-    layoutManager.orientation = RecyclerView.HORIZONTAL
-    recyclerView.layoutManager = layoutManager
-    recyclerView.setHasFixedSize(true)
-
-    val glideRequests = GlideApp.with(this)
-    adapter = RecyclerAdapter(requireContext(), glideRequests)
-    val preloader = RecyclerViewPreloader(glideRequests, adapter, adapter, 3)
-    recyclerView.addOnScrollListener(preloader)
-    recyclerView.adapter = adapter
-    return result
+  @Composable
+  private fun rememberGlidePainter(
+    requestManager: RequestManager,
+    model: Any?,
+    requestBuilder: RequestBuilder<Drawable>,
+    size: ResolvableGlideSize,
+  ): GlidePainter =
+    // TODO(judds): Calling onRemembered here manually might make a minor improvement in how quickly
+    //  the image load is started, but it also triggers a recomposition. I can't figure out why it
+    //  triggers a recomposition
+    remember { GlidePainter(requestManager, model, requestBuilder, size) }
+
+  // This class is inspired by a similar implementation in the excellent Coil library
+  // (https://github.com/coil-kt/coil), specifically:
+  // https://github.com/coil-kt/coil/blob/main/coil-compose-base/src/main/java/coil/compose/AsyncImagePainter.kt
+  @Stable
+  private inner class GlidePainter(
+    private val requestManager: RequestManager,
+    private val model: Any?,
+    private val requestBuilder: RequestBuilder<Drawable>,
+    private val size: ResolvableGlideSize,
+  ) : Painter(), RememberObserver {
+    private var alpha: Float by mutableStateOf(DefaultAlpha)
+    private var colorFilter: ColorFilter? by mutableStateOf(null)
+    private var delegate: Painter? by mutableStateOf(null)
+    private var rememberScope: CoroutineScope? = null
+
+    override val intrinsicSize: Size
+      get() = delegate?.intrinsicSize ?: Size.Unspecified
+
+    override fun DrawScope.onDraw() {
+      delegate?.apply { draw(size, alpha, colorFilter) }
+    }
+
+    override fun onAbandoned() {
+      clear()
+      (delegate as? RememberObserver)?.onAbandoned()
+    }
+
+    override fun onForgotten() {
+      clear()
+      (delegate as? RememberObserver)?.onForgotten()
+    }
+
+    override fun onRemembered() {
+      if (rememberScope != null) {
+        return
+      }
+      val scope = CoroutineScope(SupervisorJob() + Dispatchers.Main.immediate)
```

Comment:
This class should probably accept the coroutine scope as a constructor parameter, and then the scope should be obtained from the composable using `rememberCoroutineScope`. Then the scope will take advantage of structured concurrency and you don't have to manage it yourself. E.g.

```kotlin
// Above, in rememberGlidePainter.
val scope = rememberCoroutineScope()
remember(…) { GlidePainter(…, scope) }

private inner class GlidePainter(
  …,
  scope: CoroutineScope
) {
  private val scope = scope + SupervisorJob(parent = scope.coroutineContext.job) + Dispatchers.Main.immediate
}
```

Reply:
Done.

Just like with the TODO around calling `.onRemembered, in the new version of the code I see two calls to SizedGlideImage if I launch the image load in `init {}`, but one if I launch it in `onRemembered`:

```
2022-08-12 16:31:51.986 24711-24711/com.bumptech.glide.samples.gallery D/TEST: GlidePreloader
2022-08-12 16:31:51.986 24711-24711/com.bumptech.glide.samples.gallery D/TEST: rememberGlidePreloader
2022-08-12 16:31:52.141 24711-24711/com.bumptech.glide.samples.gallery D/TEST: GlidePreloader
2022-08-12 16:31:52.141 24711-24711/com.bumptech.glide.samples.gallery D/TEST: rememberGlidePreloader
2022-08-12 16:31:52.160 24711-24711/com.bumptech.glide.samples.gallery D/TEST: GlideImage: content://media/external/file/759
2022-08-12 16:31:52.161 24711-24711/com.bumptech.glide.samples.gallery D/TEST: SizedGlideImage: content://media/external/file/759, 2050449207
2022-08-12 16:31:52.165 24711-24711/com.bumptech.glide.samples.gallery D/TEST: Start load for content://media/external/file/759 into com.bumptech.glide.samples.gallery.HorizontalGalleryFragment$glideRequestFlow$1$target$1@e7d60c2
2022-08-12 16:31:52.176 24711-24711/com.bumptech.glide.samples.gallery D/TEST: collect
2022-08-12 16:31:52.184 24711-24711/com.bumptech.glide.samples.gallery D/TEST: layout!
2022-08-12 16:31:52.275 24711-24711/com.bumptech.glide.samples.gallery D/TEST: collect
2022-08-12 16:31:52.285 24711-24711/com.bumptech.glide.samples.gallery D/TEST: SizedGlideImage: content://media/external/file/759, 2050449207
2022-08-12 16:31:52.288 24711-24711/com.bumptech.glide.samples.gallery D/TEST: layout!
2022-08-12 16:31:52.314 24711-24711/com.bumptech.glide.samples.gallery D/TEST: collect
2022-08-12 16:31:52.318 24711-24711/com.bumptech.glide.samples.gallery D/TEST: layout!



2022-08-12 16:32:44.052 24816-24816/com.bumptech.glide.samples.gallery W/samples.galler: Accessing hidden method Landroid/content/pm/ApplicationInfo;->isSystemApp()Z (blacklist,test-api, linking, denied)
2022-08-12 16:32:44.881 24816-24816/com.bumptech.glide.samples.gallery D/TEST: GlidePreloader
2022-08-12 16:32:44.881 24816-24816/com.bumptech.glide.samples.gallery D/TEST: rememberGlidePreloader
2022-08-12 16:32:45.035 24816-24816/com.bumptech.glide.samples.gallery D/TEST: GlidePreloader
2022-08-12 16:32:45.035 24816-24816/com.bumptech.glide.samples.gallery D/TEST: rememberGlidePreloader
2022-08-12 16:32:45.051 24816-24816/com.bumptech.glide.samples.gallery D/TEST: GlideImage: content://media/external/file/759
2022-08-12 16:32:45.052 24816-24816/com.bumptech.glide.samples.gallery D/TEST: SizedGlideImage: content://media/external/file/759, 2050449207
2022-08-12 16:32:45.062 24816-24816/com.bumptech.glide.samples.gallery D/TEST: Start load for content://media/external/file/759 into com.bumptech.glide.samples.gallery.HorizontalGalleryFragment$glideRequestFlow$1$target$1@7720909
2022-08-12 16:32:45.072 24816-24816/com.bumptech.glide.samples.gallery D/TEST: collect
2022-08-12 16:32:45.073 24816-24816/com.bumptech.glide.samples.gallery D/TEST: layout!
2022-08-12 16:32:45.165 24816-24816/com.bumptech.glide.samples.gallery D/TEST: collect
2022-08-12 16:32:45.176 24816-24816/com.bumptech.glide.samples.gallery D/TEST: layout!
2022-08-12 16:32:45.203 24816-24816/com.bumptech.glide.samples.gallery D/TEST: collect
2022-08-12 16:32:45.209 24816-24816/com.bumptech.glide.samples.gallery D/TEST: layout!
```

Any thoughts on why starting the image load immediately triggers a recomposition, but waiting for `onRemembered` does not? The Flow that's called here should immediately produce a null Drawable, but if that's the cause, I'd expect recomposition regardless of which place I launch it.